### PR TITLE
Number System Conversion Programs in C++

### LINKED ISSUE #2188: Number System conversion Programs
### Description

Program to convert from any base to Decimal in cpp.
Program to convert from decimal to any other base in cpp. 

### Domain

C/CPP

### Type of Contribution

Documentation

### Code of Conduct

- [X] I follow [Contributing Guidelines](https://github.com/girlscript/winter-of-contributing/blob/main/.github/CONTRIBUTING.md) & [Code of conduct](https://github.com/girlscript/winter-of-contributing/blob/main/.github/CODE_OF_CONDUCT.md) of this project.

### PR DESCRIPTION
**Pull request by Nandani-koli**
<hr>

## Description 📜

Program to convert from any base to Decimal in cpp.
Program to convert from decimal to any other base in cpp.
Fixes #2188

<hr>

## Type of change 📝

- [x] Documentation (Content Creation in the form of codes or tutorials)

<hr>

## Domain of Contribution 📊

- [x] C/CPP

<hr>

## Checklist ✅

- [x] I follow [Contributing Guidelines](https://github.com/girlscript/winter-of-contributing/blob/main/.github/CONTRIBUTING.md) & [Code of conduct](https://github.com/girlscript/winter-of-contributing/blob/main/.github/CODE_OF_CONDUCT.md) of this project.
- [x] I have performed a self-review of my own code or work.
- [x] I have commented my code, particularly in hard-to-understand areas.
- [x] My changes generates no new warnings.
- [x] I'm GWOC'21 contributor

<hr>

### CODE REVIEW

**Comment by deeqakkk** [comment]
> @deeqakkk Sir, actually someone already created the theory description of types of number system and their conversion in the same folder [here](https://github.com/girlscript/winter-of-contributing/blob/C_CPP/C_CPP/Number%20System/num_sys.md) That's why I created issue for practical programs. Is this the changes you want ? or I misunderstood ? if so, i am sorry, please clarify

Thank you for pinging me on discord, and sorry for the delay.
The change I want you to do is, just add a single line descripton with an e.g so that if someone is reading your content then he/she need not switch between files to see the code and theory separately.

**Comment by deeqakkk** [comment]
> > @deeqakkk Sir, actually someone already created the theory description of types of number system and their conversion in the same folder [here](https://github.com/girlscript/winter-of-contributing/blob/C_CPP/C_CPP/Number%20System/num_sys.md) That's why I created issue for practical programs. Is this the changes you want ? or I misunderstood ? if so, i am sorry, please clarify
> 
> Thank you for pinging me on discord, and sorry for the delay. The change I want you to do is, just add a single line descripton with an e.g so that if someone is reading your content then he/she need not switch between files to see the code and theory separately.

or instead of all these just add a link to theory files in your code.

**Comment by Nandani-koli** [comment]
@deeqakkk ok sir  I got it, I will try to do the best possible changes...

File: C_CPP/Number System/Conversion-Programs/Decimal-to-other-base-conversion.md [hunk 1]
```diff
@@ -70,6 +76,8 @@ int main()
 ![octal](https://user-images.githubusercontent.com/70843941/138641591-c092f98d-1550-4a0f-bf86-759698948025.png)
 
 ## Decimal To HexaDecimal
+*In this conversion the decimal number is divided by 16 and remainder will be the digit of equivalent hexadecimal number. these remainders are arranged in reverse (bottom to top) order.*
+*(Note:- if remainder is 10, 11, 12, 13, 14, 15 then in hexadecimal it will be A,B,C,D,E,F respectively.*
```
Comment (deeqakkk) [inline C_CPP/Number System/Conversion-Programs/Decimal-to-other-base-conversion.md:80]: Nandani, close this bracket which started at Note:-.

**Review by deeqakkk — CHANGES_REQUESTED** [review]
just make this one change.
@Suvraneel it's good to be merged after the change.

File: C_CPP/Number System/Conversion-Programs/Decimal-to-other-base-conversion.md [hunk 2]
```diff
@@ -0,0 +1,119 @@
...
+    while (n != 0)
+    {
+        rem = n%2;      //store the reminder 
+        n /= 2;         // divide num by base(2), store the Quotient
+        ans = ans + rem * i ;  // store the answer 
+        i *= 10;  // increment placevalue 
+    }
+    return ans ;
+
+}
```
Comment (Suvraneel) [inline C_CPP/Number System/Conversion-Programs/Decimal-to-other-base-conversion.md:27]:
```suggestion
    return ans ;
}
```

File: C_CPP/Number System/Conversion-Programs/Decimal-to-other-base-conversion.md [hunk 3]
```diff
@@ -0,0 +1,119 @@
...
+int main()
+{
+    //Take input as decimal number 
+    int num;
+    cout << "enter decimal number:- ";
+    cin >> num;
+    cout<< decimalToBinary(num);   // function call
+}
+```
+## OUTPUT:-
```
Comment (Suvraneel) [inline C_CPP/Number System/Conversion-Programs/Decimal-to-other-base-conversion.md:37]:
```suggestion
## OUTPUT  
```

File: C_CPP/Number System/Conversion-Programs/Decimal-to-other-base-conversion.md [hunk 4]
```diff
@@ -0,0 +1,119 @@
...
+
+    int num;
+  
+    //Take input as decimal number 
+    cout << "enter decimal number:- ";
+    cin >> num;
+    cout<< decimalToOctal(num);         //function call
+}
+```
+## OUTPUT:-
```
Comment (Suvraneel) [inline C_CPP/Number System/Conversion-Programs/Decimal-to-other-base-conversion.md:75]:
```suggestion
## OUTPUT  
```

File: C_CPP/Number System/Conversion-Programs/Decimal-to-other-base-conversion.md [hunk 5]
```diff
@@ -0,0 +1,119 @@
...
+int main()
+{
+    //Take input as decimal number 
+    int num;
+    cout << "enter decimal number:- ";
+    cin >> num;
+    decimalToHexadecimal(num);      //function call
+}
+```
+## OUTPUT:-
```
Comment (Suvraneel) [inline C_CPP/Number System/Conversion-Programs/Decimal-to-other-base-conversion.md:117]:
```suggestion
## OUTPUT  
```

File: C_CPP/Number System/Conversion-Programs/Other-base-to-Decimal-Conversion.md [hunk 6]
```diff
@@ -0,0 +1,108 @@
...
+    while (n > 0)                           //loop till number greater than 0
+    {
+        int lastdigit = n%10;                 //get the last digit of number
+        decimal += x * lastdigit;              //add in answer
+        x *= 8;                                 //increase the power by base(8)
+        n/= 10;                                 //remove the last digit 
+    }
+    return decimal;                             //return answer
+}
+int main()
```
Comment (Suvraneel) [inline C_CPP/Number System/Conversion-Programs/Other-base-to-Decimal-Conversion.md:59]:
```suggestion
}

int main()
```

File: C_CPP/Number System/Conversion-Programs/Other-base-to-Decimal-Conversion.md [hunk 7]
```diff
@@ -0,0 +1,108 @@
...
+{
+    int num;
+    cout << "enter octal number:- ";                //Input octal number 
+    cin >> num;
+
+    cout<<octalToDecimal(num);                      //call a function
+}
+```
+# OUTPUT:-
```
Comment (Suvraneel) [inline C_CPP/Number System/Conversion-Programs/Other-base-to-Decimal-Conversion.md:68]:
```suggestion
## OUTPUT  
```
Wrong level of heading

File: C_CPP/Number System/Conversion-Programs/Other-base-to-Decimal-Conversion.md [hunk 8]
```diff
@@ -0,0 +1,108 @@
...
+        }
+        else if (n[i] >= 'A' && n[i] <= 'F')          //else digit is a char 
+        {
+            sum += x * (n[i] - 'A' + 10);             //add in answer 
+        }
+        x = x * 16;                                    //increase power by base (16)
+    }
+    return sum;                                       //return answer
+}
+int main()
```
Comment (Suvraneel) [inline C_CPP/Number System/Conversion-Programs/Other-base-to-Decimal-Conversion.md:97]:
```suggestion
}

int main()
```

File: C_CPP/Number System/Conversion-Programs/Other-base-to-Decimal-Conversion.md [hunk 9]
```diff
@@ -0,0 +1,108 @@
...
+{
+    string num;
+    cout << "enter hexaDecimal number:- ";                    //input hexadecimal number
+    cin >> num;
+
+    cout<< hexaToDecimal(num);                                //function call
+}
+```
+# OUTPUT:-
```
Comment (Suvraneel) [inline C_CPP/Number System/Conversion-Programs/Other-base-to-Decimal-Conversion.md:106]:
```suggestion
## OUTPUT  
```

**Review by Suvraneel — CHANGES_REQUESTED** [review]

**Comment by Nandani-koli** [comment]
@deeqakkk @Suvraneel Thank you so much sir.

**Comment by Suvraneel** [comment]
@Nandani-koli  LGTM  :octocat:  ❤️  Nice work 🚀

**Pull request merged**